src/rgw/rgw_rest.cc

```diff
@@ -241,7 +241,7 @@ static bool str_ends_with(const string& s, const string& suffix, size_t *pos)
     *pos = p;
   }
 
-  return s.compare(p, len, suffix) == 0;
+  return boost::algorithm::iends_with(suffix, s);
```

Comment:
i'm seeing failures from the s3website tests, and i think it's because these arguments are in the wrong order?

`iends_with("test123", "123")` returns true, where `iends_with("123", "test123")` returns false

Reply:
testing a fix in https://github.com/ceph/ceph/pull/30221